proxy/src/auth/backend.rs

```diff
@@ -143,13 +151,55 @@ async fn auth_quirks(
     // Currently, we use it for websocket connections (latency).
     if allow_cleartext {
         // Password will be checked by the compute node later.
-        return hacks::cleartext_hack(api, extra, creds, client).await;
+        return hacks::cleartext_hack(client).await;
     }
 
     // Finally, proceed with the main auth flow (SCRAM-based).
     classic::authenticate(api, extra, creds, client, config).await
 }
 
+/// True to its name, this function encapsulates our current auth trade-offs.
+/// Here, we choose the appropriate auth flow based on circumstances.
+async fn auth_quirks(
+    api: &impl console::Api,
+    extra: &ConsoleReqExtra<'_>,
+    creds: &mut ClientCredentials<'_>,
+    client: &mut stream::PqStream<impl AsyncRead + AsyncWrite + Unpin>,
+    allow_cleartext: bool,
+    config: &'static AuthenticationConfig,
+) -> auth::Result<AuthSuccess<CachedNodeInfo>> {
+    let auth_stuff = auth_quirks_creds(api, extra, creds, client, allow_cleartext, config).await?;
+
+    let mut num_retries = 0;
+    let mut node = loop {
```

Comment:
I noticed that we have similar codeblocks in different places. Should we consider to make a method/macro to avoid copy-pasting?

Reply:
It used to be worse 😅 there's currently only 2 of these retry loops in ths proxy. One here, and one in connect_to_compute.

We could probably abstract it into a wake_compute_with_retries but I think I'd rather make that change in a different PR